src/compiler/checker.ts

```diff
@@ -32023,14 +32045,48 @@ namespace ts {
         }
 
         function checkInExpression(left: Expression, right: Expression, leftType: Type, rightType: Type): Type {
-            if (leftType === silentNeverType || rightType === silentNeverType) {
-                return silentNeverType;
+            if (isPrivateIdentifier(left)) {
+                const privateId = left;
+                const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(privateId.escapedText, privateId);
+                if (lexicallyScopedSymbol === undefined) {
+                    if (!getContainingClass(privateId)) {
+                        error(privateId, Diagnostics.Private_identifiers_are_not_allowed_outside_class_bodies);
+                    }
+                    else {
+                        const suggestion = getSuggestedSymbolForNonexistentProperty(privateId, rightType);
+                        if (suggestion) {
+                            const suggestedName = symbolName(suggestion);
+                            error(privateId, Diagnostics.Cannot_find_name_0_Did_you_mean_1, diagnosticName(privateId), suggestedName);
+                        }
+                        else {
+                            error(privateId, Diagnostics.Cannot_find_name_0, diagnosticName(privateId));
+                        }
+                    }
+                    return anyType;
+                }
+
+                markPropertyAsReferenced(lexicallyScopedSymbol, /* nodeForCheckWriteOnly: */ undefined, /* isThisAccess: */ false);
+                getNodeLinks(privateId.parent).resolvedSymbol = lexicallyScopedSymbol;
+                if (rightType === silentNeverType) {
+                    return silentNeverType;
+                }
+                rightType = checkNonNullType(rightType, right);

```

Comment:
You can probably move this line outside of the `if..else` and remove the line at 32080 since they do the same thing.

Reply:
Done :)